.github/workflows/auto-review-and-pr-summary.yml

```
@@ -0,0 +1,84 @@
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  generate_and_update:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)